utils/httpUtils.ts

```diff
@@ -91,3 +147,42 @@ export const notFound = (message: string) => {
     message: message,
   });
 };
+
+export const getFeatureFlags = async (context: any) => {
+  try {
+    const response = await fetch(process.env.API_URL + "/users/feature_flags", {
+      method: "GET",
+      headers: {
+        Cookie: context.request.headers.get("cookie"),
+        "Content-Type": "application/json",
+      },
+    });
+    return await response.json();
+  } catch (e) {
+    return Promise.reject(e.response);
+  }
+};
+
+export const getFeatureFlagsFromRequest = (context) => {
+  return context.request.headers.get("readFromNextGen");
+};
+
+export const isNextGenEnabled = async (context) => {
```

Comment:
nit: could we rename this to something like shouldReadFromNextGen? The current name may be too generic, in case we end up needing to creating / check any other next gen related feature flags.

Specifically, I could imagine that we end up needing additional feature flags around writes and/or reading for particular features.